org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/res/loader/DynamicResourceInfo.java

```diff
@@ -106,6 +108,31 @@ public static DynamicResourceInfo fromPath(HttpServletRequest req, String path)
     return new DynamicResourceInfo(uiSession, components.getAdapterId(), components.getFilename());
   }
 
+  /**
+   * @param uiSessionResolver
+   *          non-null UI session ID -> {@link IUiSession} resolver
+   * @param jsonAdapterResolver
+   *          non-null json adapter ID -> {@link IJsonAdapter} resolver
+   * @param path
+   *          decoded path (non url-encoded)
+   */
+  public static DynamicResourceInfo fromPath(Function<String/*uiSessionId*/, IUiSession> uiSessionResolver, Function<String/*id*/, IJsonAdapter> jsonAdapterResolver, String path) {
```

Comment:
`uiSessionResolver` is not really used. It is called and the result is validated (matching ids), but the resulting `IUiSession` is not used. Therefore, delete the parameter or use it to validate the `IJsonAdapter`.